backend/src/test/java/de/ffmjava/capstone/backend/horses/HorseServiceTest.java

```diff
@@ -23,4 +25,43 @@ void getAllHorses() {
 
         assertEquals(expected, actual);
     }
-}
\ No newline at end of file
+
+    @Test
+    void addNewHorse_AndExpectHorse() {
+        Horse newHorse = new Horse(null, "name", "owner", null);
+
+        doReturn(newHorse.withId("1")).when(mockRepository).save(any());
+
+        Horse actual = service.addNewHorse(newHorse);
+
+        Horse expected = newHorse.withId("1");
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void deleteHorse_AndExpectSuccess() {
+        String idToDelete = "1";
+        when(mockRepository.existsById(idToDelete)).thenReturn(true);
+        doNothing().when(mockRepository).deleteById(idToDelete);
+
+        assertTrue(service.deleteHorse(idToDelete));
+        verify(mockRepository).existsById(idToDelete);
+    }
+
+    @Test
+    void deleteHorse_AndExpectException_404() {
+        String idToDelete = "1";
+        ResponseStatusException expectedException = new ResponseStatusException(HttpStatus.NOT_FOUND, "Kein Eintrag für die gegebene ID gefunden");
+        when(mockRepository.existsById(idToDelete))
+                .thenReturn(false);
+        doNothing().when(mockRepository).deleteById(idToDelete);
+        try {
+            service.deleteHorse(idToDelete);
+            fail();
+        } catch (ResponseStatusException e) {
+            assertEquals(expectedException.getMessage(), e.getMessage());
```

Comment:
if you only need the message its sufficient to only declare the message string...